### ISSUE TITLE
When running rubocop in a PR to ensure the code style is valid it shows offence "Class has too many lines. [321/320]"

### ISSUE THREAD

**Issue by funnel20** [fastlane/fastlane#11104]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
The [PR template](https://github.com/fastlane/fastlane/blob/master/.github/PULL_REQUEST_TEMPLATE.md) shows to run rubocop to ensure the code style is valid:
```
bundle exec rubocop -a
```

When I do this for my PR it shows:
```
Offenses:

frameit/lib/frameit/editor.rb:2:3: C: Class has too many lines. [321/320]
  class Editor ...
  ^^^^^^^^^^^^

973 files inspected, 1 offense detected
```

The solution by a team member is to use a trivial work-around by adding comments in the Class to disable the check:
https://stackoverflow.com/questions/40221797/rails-rubocop-disable-class-has-too-many-lines-error

I think it's good practise to have checks, but they should be meaningful.
A max limit of **320** code lines for a class seems not much.
Is it possible to reevaluate this value and increase it, instead of forcing coders to disable the check on files?

##### Complete output when running fastlane, including the stack trace and command used

[INSERT OUTPUT HERE]

### Environment

[INSERT OUTPUT HERE]

**Comment by ohayon** [comment]
Hey @funnel20, I think the preference would be to refactor some code out into another class so that they arent getting too big and too responsible for so many things. Is there an opportunity to do that with your PR? How many lines over 320 are you?

**Comment by funnel20** [comment]
1 line

**Comment by ohayon** [comment]
Is there a empty new line you can delete just to fix this problem and not worry about it at the moment. Classes over 320 lines definitely are too long, and while it may seem arbitrary, we do have to have _some_ checks here.

**Comment by funnel20** [comment]
No sorry, all lines are functional. I expect the check doesn't count comments.

One of the drawbacks of this check is that the fixer (me) is now the problem owner of a file that is too large. While I only modified a fraction of the file, to solve an issue.

**Comment by ohayon** [comment]
Really sorry about that @funnel20, I know some of the process can be annoying. What would your ideal situation be for this issue?

**Comment by funnel20** [comment]
I'm aware of guidelines to limit the lines of code _within functions_ (or methods, don't know what the nomenclature is in Ruby). That makes sense, because for readability and testability.

But I don't understand why a class should be limited. Moving parts to another class, doesn't always contribute to a better readability.

Let's say I create a ViewController in macOS with 32 buttons. Your rule then limits that every button can have 10 lines of code on average. 2 lines are already required to define each button callback, resulting in 8 functional lines per button. Will this check result in better code? I don't believe it, because I have to open dozens of other classes to view the logic behind  the buttons.

Another example; the _header_ of Apple's own [UIViewController](https://github.com/nst/iOS-Runtime-Headers/blob/master/Frameworks/UIKit.framework/UIViewController.h) contains already **1,700** lines. Let alone the _implementation_ file. As a professional software company tis proofs that they don't use this check, or have the limit at a significant higher level.

The current check restricts freedom and creativity and jeopardises readability:
- One solution is to merge different lines of code, which creates only 1 line and let's the check pass. But this line becomes unreadable.
- Another solution is to disable the check by adding the comment. From an architect perspective this makes the entire check obsolete.
- If a developer sees a countdown of remaining lines while writing code, he might be forced to cut corners. In the aforementioned issue that I solved, this might have been the case since just a single `.crop` command was used. 
Super efficient in lines of code, but totally **undesired side effects** for the end result.

As long as I don't know what the rational for your existing check is, and why it's limited to 320 I can't say what to change. But I hope my concerns are valid and understandable.

**Comment by ohayon** [comment]
I totally understand your point and while I think there are nicer ways to handle 32 buttons all in one view controller rather than using container view controllers containing smaller, child view controllers in iOS, Ruby is fundamentally very different than Swift and even more so than ObjC. The 320 line limit assumes (probably, more often than not, correctly) that if the class is that big *likely* there are too many responsibilities shared by that class and things could be encapsulated in some smaller classes.

All of this being said, we deeply care for our community and want to make our contributors as happy as possible. What would be your ideal outcome of this situation and we can help you get there 😄

**Comment by funnel20** [comment]
Since you're the masters of Ruby, I have no value here.
This Class contains a lot of helper related methods, like getting settings from the Framefile, loading fonts, etc. In this case I would suggest to make a new Class and move these methods to it, so the current class will stay within the max line limit.

What I don't want to hear is "as a work-around just disable the check". 
Or "we don't accept the PR because there are too many lines". Yeah, that's because other colleagues wrote those 310 lines before me.

In my ideal world the fixer of an issue who just hits this limit, is not responsible to solve this situation.
So someone from the team should accept the PR and alter the class architecture afterwards.
This will result in a win-win:

1. The fixer is satisfied that his fix is accepted, while he doesn't have to break his head about architecture. So he keeps motivated to contribute again on small portions of the code, instead of having full responsibility of the software architecture.
2. The team stays in control about the architecture, because they are likely the pro and know all the ins and outs.

**Comment by ohayon** [comment]
Hey, @funnel20, again, I totally understand the frustration that this length check is causing. And I understand that those two steps you listed make a lot of sense. 

I hope that you can understand that we do need to have some guard rails in place to make sure that our code base doesn't get out of control. Given the fact that we need to have some checks in place, this situation is bound to happen. 

I will submit a commit on top of your PR to handle the line number issue so that you wont have to worry about changing any architecture or refactoring anyone else's code and then we can make sure we set you up for success the next time (we hope) you want to contribute something! 😄 🚀

**Comment by funnel20** [comment]
@ohayon Thanks for reading my concerns. 
As mentioned, I don't have problems with checks, but there needs to be an acceptable workflow to make them pass.
Thanks for making the change to my PR by reducing the code with 1 line to let the check pass.
Although I hope that you agree that the inline solution doesn't contribute to readability 😈.

Should we create an issue/task for refactoring this Class to prevent any future failing line check?

**Comment by ohayon** [comment]
I definitely agree that the inline solution is not ideal! 

We have created a task for our team to follow up with a refactor to the class so that we can have more flexibility (we don't typically use GitHub issues for our task tracking, rather we only use it for handling user support). I wonder if rubocop can warn us when classes are getting close to 320 so we have a chance to refactor things before this type of frustration is imposed on anyone?

All of this being said, we would really love to get your change merged in, and then we can go ahead with a refactor, just waiting to see if we can find a solution to the CLA issue. 😄 

If you don't mind, I'd like to close this issue as we have created the task to refactor the code and discuss the situation internally and it is not surfacing an actual bug in the fastlane code which is what we ideally use GitHub Issues for. Let me know your thoughts! 🐙
